Automattic/wp-calypso
# Move "Make Primary" to domains list

Fixes #2531.

This add the Make Primary button to Domains -> List. (It does not remove the old way – let's watch adoption first).

### Screenshots

#### Primary Domain Mode Disabled (default)
![image](https://cloud.githubusercontent.com/assets/991022/13654533/cc627f80-e60d-11e5-9989-5816f143bedb.png)

#### Primary Domain Mode Enabled
![image](https://cloud.githubusercontent.com/assets/991022/13654550/e59d16a4-e60d-11e5-8338-8674b35ad86e.png)

#### Click on domain
![image](https://cloud.githubusercontent.com/assets/991022/13654562/f92de810-e60d-11e5-84d9-510500a71095.png)

#### Successful change
![image](https://cloud.githubusercontent.com/assets/991022/13654579/087f49b2-e60e-11e5-9669-70e356cc30ec.png)

#### Successful undo

![image](https://cloud.githubusercontent.com/assets/991022/13676347/531fd220-e69a-11e5-9540-970064aba944.png)


#### Fail change
![image](https://cloud.githubusercontent.com/assets/991022/13654737/d6a0a1ba-e60e-11e5-91f1-05777ad4034a.png)


### Testing
- Go to Domains -> List
- Click "Change Primary"
- Click any of the domains
- Wait
- Confirm you've changed the primary domain
- Confirm you the URL changes at the same time with the success notice
- Confirm you can navigate the site without problems (back button will break for old URLs – it's a known issue without a feasible fix)
- Confirm undo works.
- Changing primary domain name is a fragile thing to do since we rely on URLs to determine the active site and we are actively changing the canonical URL. So look out for stuff.

### Things to look for
- Event names for tracking

/cc: @klimeryk @aidvu 
/cc: @mikeshelton1503 @breezyskies
